pkgs/by-name/va/vanguards/package.nix

```diff
@@ -0,0 +1,37 @@
+{ pkgs }:
+let
+  vgpkgs = import (pkgs.fetchFromGitHub {
+    # required version for stem and ipaddress to work
+    owner = "NixOS";
+    repo = "nixpkgs";
+    rev = "f539a47dabe6fd614179ee7df12d596bdab1dff9";
+    sha256 = "sha256-wXhy+5YmZRaUUaZKnuRIfsOdHsPBQq2ewdMFDS2uRsw=";
+  }) { };
+in
+vgpkgs.python2Packages.buildPythonPackage rec {
+  pname = "vanguards";
+  version = "0.3.1";
+  meta = {
+    maintainers = [ "ForgottenBeast" ];
+    mainProgram = "vanguards";
+    license = "MIT";
+    description = ''
+      Runs alongside tor and interacts with its control port
+      in order to protect and alert against guard node attacks on hidden services'';
+  };
+  doCheck = false;
+  src = pkgs.fetchFromGitHub {
+    owner = "mikeperry-tor";
+    repo = pname;
+    rev = "c3961ac40ca0bce67f79bc76021f5817730033b8";
+    sha256 = "sha256-y5WwDLn2asYcA5hTl++UVeH5KZ8VRP4sMIjRv9y7GVE=";
+  };
+  propagatedBuildInputs = with vgpkgs; [
```

Comment:
Why is this dangling on its own? This derivation is really hard to read.

Reply:
I am not sure I understand what you mean by "dangling on its own". The source repository does not have a requirements list or poetry equivalent so I had to explicitly add both of those libraries this way.

If I remove this bit the build fails with the following error message (linked with the removal of ipaddress I should think)

ERROR: Could not find a version that satisfies the requirement ipaddress>=1.0.17; python_version < "3" (from vanguards==0.3.1) (from versions: none)
ERROR: No matching distribution found for ipaddress>=1.0.17; python_version < "3" (from vanguards==0.3.1)

Reply:
@Frontear I'm interested in suggestions regarding the issue with older python 2.7 and related libraries as it seems to be the last pending item

Reply:
You can override the versions used using overrideAttrs. However I would advise against adding new dependencies on Python 2 in nixpkgs.

Reply:
Thank you @FliegendeWurst , I will do that!
In this case the software wasn't updated for Python 3, what would be the best course of action, attempt a fork?

Reply:
@FliegendeWurst  Not sure how to proceed the "ipaddress" lib, since it's been removed, not upgraded after this nixpkgs version. What should I do the override with?

Reply:
> In this case the software wasn't updated for Python 3, what would be the best course of action, attempt a fork?

Yes. Though I think you can just pick up the Ubuntu patch for Python 3. https://git.launchpad.net/ubuntu/+source/vanguards/tree/debian/patches

Reply:
ipaddress is no longer required. It was merely a backport of 3.x standard library functions.

Reply:
Thanks. I am currently testing the Ubuntu patches. There seems to be an issue with the stem library, I see if I can pin the latest working version